INSTALL

```diff
@@ -12,6 +12,16 @@ without warranty of any kind.
 Basic Installation
 ==================
 
+   This project is configured with autoconf, so, if building from a
+clean git clone of the project, i.e., where autoconf generated files 
+are not already provided, they need to be created.  To do this, install
+autoconf tools: e.g., 'sudo apt install automake',  and run:
+
+	aclocal -I .; autoheader; autoconf; automake --add-missing -c
+
+That will create the config.h.in, aclocal.m4, configure, and Makefile.in
+files necessary to proceed to the next step.
+
    Briefly, the shell commands `./configure; make; make install' should
```

Comment:
As configure is removed, this is not accurate anymore. Could you update the instructions on how to build/install the package? Thanks.

Reply:
not sure I understand.  ./configure is removed, so I updated the instructions to include how to create ./configure, before where they say to use ./configure (line starting with "Briefly,").  Running ./configure is still needed in order to build. Do you want me to replace the word "Briefly," with "Then," or something?

Reply:
Ahh, I see, the first paragraph is the first step.

With this change, the original 3-steps build-install process is expanded are extedned to 7 steps for users who downloaded the package for the first time. It would be nice to have the first few steps into a setup.sh, so that people just need to run setup.sh; make; make install

Another question is, with this change, if I do make dist-clean, will it remove all generated file too?

Thanks for working on this!

Reply:
So you want a new file, setup.sh to contain basically one line ("aclocal -I .; autoheader; autoconf; automake --add-missing -c"), and then a plain "./configure" but not "make; make install"?  Users don't naturally expect that, and I have personally seen the autoconf instructions laid out the way I did in the instructions, before.  setup.sh it doesn't allow for open customization: installers would want to add a switch or two to ./configure as they run it and examine its output...so, I'm not sure if a new setup.sh is the way to go...

wrt 'make distclean', no, it removes .deps/*, Makefile, config.h, config.log, and config.status.  It leaves these autoconf (automake?) -generated files behind:

Makefile.in
aclocal.m4
autom4te.cache/output.0
autom4te.cache/output.1
autom4te.cache/output.2
autom4te.cache/requests
autom4te.cache/traces.0
autom4te.cache/traces.1
autom4te.cache/traces.2
compile
config.h.in
configure
depcomp
install-sh
missing

...which is what matches the old INSTALL instructions that start with 'run ./configure; make; make install'.

I don't see how to change the 'make distclean' behaviour, since the distclean rule is only in the autoconf-generated Makefile.

I'm far from an autoconf expert, are you saying we should leave these files that make distclean leaves behind? I thought we'd still get issues like #60 which says "The aclocal.m4 is missing, so we need to regenerate every time. If users have a different version of automake, it complains."  So I thought if we wiped out all the build files like aclocal.m4, there wouldn't be a different version of automake to complain about.  I asked that user for feedback on this pull request but didn't get any, and prior commits (mainly from Gentoo maintainers, see e.g., commit e21776e "delete autogenerated files") has been to remove all these build artifacts, including ./configure, etc.

FWIW, I've been using 'git clean -f -x' as an equivalent to 'make distclean', and/or 'make maintainer-clean'.  The latter rule is apparently deprecated from recent automake versions...